test/functional/interface_rpc.py

```diff
@@ -18,6 +20,20 @@ def expect_http_status(expected_http_status, expected_rpc_code,
         assert_equal(exc.error["code"], expected_rpc_code)
         assert_equal(exc.http_status, expected_http_status)
 
+got_exceeded_error = False
+def test_work_queue_getblock(node):
+    global got_exceeded_error
+    for _ in range(400):
+        if got_exceeded_error:
+            break
+        try:
+            node.cli('getrpcinfo').send_cli()
+        except subprocess.CalledProcessError as e:
+            if e.output != 'error: Server response: Work queue depth exceeded\n':
+                raise AssertionError("Unexpected cli error: :{}:".format(e.output))
+            got_exceeded_error = True
+            break
```

Comment:
This 400 is, of course, arbitrary, as is the number of threads this test starts (5, see below). These numbers are the result of simple trial-and-error. We want each thread to do enough work that it's still running by the time all 5 threads are started. Still, the test only takes about 1.3 seconds to run on my system. Increasing the number of threads allows the iteration count here to be reduced, but I don't want to start too many threads, because each one probably uses a nontrivial amount of resources and I don't want to overburden the CI workers.